scipy/optimize/zeros.py

```diff
@@ -206,6 +222,86 @@ def newton(func, x0, fprime=None, args=(), tol=1.48e-8, maxiter=50,
     raise RuntimeError(msg)
 
 
+def _array_newton(func, x0, fprime, args, tol, maxiter, fprime2,
+                  failure_idx_flag=False):
+    """
+    A vectorized version of Newton, Halley, and secant methods for arrays. Do
+    not use this method directly. This method is called from :func:`newton`
+    when ``np.isscalar(x0)`` is true. For docstring, see :func:`newton`.
+    """
+    p = np.asarray(x0)
+    failures = np.ones_like(p, dtype=bool)  # at start, nothing converged
+    if fprime is not None:
+        # Newton-Raphson method
+        for iteration in range(maxiter):
+            fder = np.asarray(fprime(p, *args))
+            zero_der = (fder == 0)
+            # stop iterating if all derivatives are zero
+            if zero_der.all():
+                break
+            fval = np.asarray(func(p, *args))
+            # Newton step
+            dp = fval / fder
+            if fprime2 is not None:
+                fder2 = np.asarray(fprime2(p, *args))
+                dp = dp / (1.0 - 0.5 * dp * fder2 / fder)
+            # use last guess where derivative is zero
+            p = np.where(zero_der, p, p - dp)
```

Comment:
Also, notice that by using `zero_der` as your mask here, you will continue to update those items that had already converged. Not that there's anything too wrong with that, since it should only improve the result further, but the return may be different than passing a single scalar for those values. At the very least it may merit a short note in the docstring? Or maybe not, don't have a strong opinion about this.

Reply:
✔️